File: 프로그래머스/0/181856. 배열 비교하기/배열 비교하기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> arr1, vector<int> arr2) {
    int sum1 = 0, sum2 = 0;
    
    if(arr1.size() != arr2.size()) { // 두 배열의 사이즈가 다를 경우
        if(arr1.size() > arr2.size()) { // 두 배열의 사이즈 비교
            return 1;
        } else {
            return -1;
        }
    } else { // 두 배열의 사이즈가 같을 경우
        for(int i=0;i<arr1.size();i++) {
            sum1 += arr1[i];
            sum2 += arr2[i];
        }
        if(sum1 > sum2) {
            return 1;
        } else if(sum1 < sum2) {
            return -1;
        } else {
            return 0;
        }
    }
}

File: 프로그래머스/0/181856. 배열 비교하기/배열 비교하기.swift
import Foundation

func solution(_ arr1:[Int], _ arr2:[Int]) -> Int {
    
    if(arr1.count != arr2.count) {
        return arr1.count > arr2.count ? 1 : -1
    } else {
        if(arr1.reduce(0, +) == arr2.reduce(0, +)) {
            return 0
        } else {
            return (arr1.reduce(0, +) > arr2.reduce(0, +)) ? 1 : -1
        }
    }
}